### ISSUE TITLE
Error metric for avoiding false positives with high scores

### ISSUE THREAD

**Issue by alejandrogomez97** [microsoft/LightGBM#6031]
Hi. I have a binary variable with two possible values, 0 or 1. It's an imbalanced problem, 98% of my data is 0 and 2% is 1. My objective is to give a list of 100 the instances of the test set which have more probabilities to be 1. 

Until now, I was facing this with a binary lightgbm classifier using the average precision metric. The problem is that using this metric i'm focusing on predicting all the data and i'm only interested in predicting correctly the 100 instances with higher scores. Intuitively, if i used a metric to train that depended on the predicted score of the algorithm I could have less false positives in the first 100 instances. How can I do so? Is there any metric that can help me solve this issue?

**Comment by jameslamb** [comment]
I see that you double-posted this here and on Stack Overflow ([link](https://stackoverflow.com/questions/76881562/train-a-ml-algorithm-to-focus-only-on-high-score-instances)).

Please do not do that.

Maintainers here also monitor the `[lightgbm]` tag on Stack Overflow. I could have been spending time preparing an answer here while another maintainer was spending time answering your Stack Overflow post, which would have been a waste of maintainers' limited attention that could otherwise have been spent improving this project. Double-posting also makes it less likely that others with a similar question will find the relevant discussion and answer from search engines.
